homeassistant/components/device_tracker/__init__.py

```diff
@@ -594,6 +648,104 @@ def mark_stale(self):
                         state.attributes[ATTR_LONGITUDE])
 
 
+class DeviceTrackerEntity(Entity):
+    """Represent a tracked device."""
+
+    @property
+    def battery(self):
+        """Return the battery level of the device."""
+        return None
+
+    @property
+    def gps_accuracy(self):
+        """Return the gps accuracy of the device."""
+        return DEFAULT_GPS_ACCURACY
+
+    @property
+    def is_connected(self):
+        """Return True if device is connected."""
+        return None
+
+    @property
+    def latitude(self):
+        """Return latitude value of the device."""
+        return None
+
+    @property
+    def location_name(self):
```

Comment:
What about making it an area and/or zone reference instead? I.e. platforms should try and map location to area/zone.  Does an arbitrary `location_name` provide value given each tracker may report something different or are there is there a schema of expected names?

Reply:
Zone is handled in `state` property.

Reply:
This is the existing interface. I've tried to keep it as much as possible.

Reply:
Yeah - Probably that's a different arch issue to deal with later.